Importing Frontdoor route causes Panic

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.0

### AzureRM Provider Version

3.27.0

### Affected Resource(s)/Data Source(s)

azurerm_cdn_frontdoor_route

### Terraform Configuration Files

```hcl
terraform {

 }

provider "azurerm" {
  features {}
  }

resource "azurerm_cdn_frontdoor_route" "shell" {
  cache {}
}
```


### Debug Output/Panic Output

```shell
Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-azurerm_v3.27.0_x5.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x4632065]

goroutine 131 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn.flattenRuleSetResourceArray(0xc0019870b0)
        github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/cdn_frontdoor_route_resource.go:617 +0xc5
github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn.resourceCdnFrontDoorRouteRead(0xc00014b900, {0x59022e0?, 0xc000598c00?})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/cdn_frontdoor_route_resource.go:375 +0xab4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x6a78e48?, {0x6a78e48?, 0xc0012f31a0?}, 0xd?, {0x59022e0?, 0xc000598c00?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00035d960, {0x6a78e48, 0xc0012f31a0}, 0xc001961380, {0x59022e0, 0xc000598c00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00011d7a0, {0x6a78e48?, 0xc0012f30e0?}, 0xc00196cd80)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:613 +0x497
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000dfb860, {0x6a78e48?, 0xc0012f29c0?}, 0xc001957e00)
        github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:747 +0x41e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x5f00000?, 0xc000dfb860}, {0x6a78e48, 0xc0012f29c0}, 0xc001957da0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002b0a80, {0x6a8aaa0, 0xc000ae0000}, 0xc000d64900, 0xc000cdaf60, 0xab76830, 0x0)
        google.golang.org/grpc@v1.47.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc0002b0a80, {0x6a8aaa0, 0xc000ae0000}, 0xc000d64900, 0x0)
        google.golang.org/grpc@v1.47.0/server.go:1620 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.47.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.47.0/server.go:920 +0x28a

Error: The terraform-provider-azurerm_v3.27.0_x5.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

Resource should be imported

### Actual Behaviour

Terraform crashes

### Steps to Reproduce

terraform init
terraform import azurerm_cdn_frontdoor_route.shell "<resource_id>"

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/cdn/cdn_frontdoor_helpers.go
+++ b/internal/services/cdn/cdn_frontdoor_helpers.go
@@ -251,7 +251,7 @@ func flattenCustomDomainActivatedResourceArray(input *[]cdn.ActivatedResourceRef
 	}
 
 	// Normalize these values in the configuration file we know they are valid because they were set on the
-	// resource... if these are modified in the portal the will all be lowercased...
+	// resource... if these are modified in the portal they will all be lowercased...
 	for _, customDomain := range *input {
 		if customDomain.ID == nil {
 			continue

--- a/internal/services/cdn/cdn_frontdoor_route_resource.go
+++ b/internal/services/cdn/cdn_frontdoor_route_resource.go
@@ -611,7 +611,7 @@ func flattenRuleSetResourceArray(input *[]cdn.ResourceReference) []interface{} {
 	}
 
 	// Normalize these values in the configuration file we know they are valid because they were set on the
-	// resource... if these are modified in the portal the will all be lowercased...
+	// resource... if these are modified in the portal they will all be lowercased...
 	// Issue: https://github.com/Azure/azure-sdk-for-go/issues/19378
 	for _, ruleSet := range *input {
 		id, _ := parse.FrontDoorRuleSetIDInsensitively(*ruleSet.ID)

--- a/internal/services/cdn/cdn_frontdoor_route_resource.go
+++ b/internal/services/cdn/cdn_frontdoor_route_resource.go
@@ -612,8 +612,9 @@ func flattenRuleSetResourceArray(input *[]cdn.ResourceReference) []interface{} {
 
 	// Normalize these values in the configuration file we know they are valid because they were set on the
 	// resource... if these are modified in the portal the will all be lowercased...
+	// Issue: https://github.com/Azure/azure-sdk-for-go/issues/19378
 	for _, ruleSet := range *input {
-		id, _ := parse.FrontDoorRuleSetID(*ruleSet.ID)
+		id, _ := parse.FrontDoorRuleSetIDInsensitively(*ruleSet.ID)
 		results = append(results, id.ID())
 	}
 

